### PR TITLE
Add in-toto attestations to rebuilderd

### PR DESCRIPTION
**Pull request by adityasaky**
Upstreaming @joyliu-q's outstanding work to add in-toto attestations to rebuilderd.

### CODE REVIEW

**Comment by adityasaky** [comment]
cc @joyliu-q

File: worker/src/rebuild.rs [hunk 1]
```diff
@@ -128,12 +131,33 @@ pub async fn rebuild(ctx: &Context<'_>, url: &str) -> Result<Rebuild> {
...
     let output_path = out_dir.join(&filename);
+
+    info!("Generating signed link");
+
+
+    let signed_link = in_toto_run(
+        &format!("rebuild {}", filename.to_str().unwrap()),
+        None,
+        &[input_path.to_str().unwrap()],
+        &[output_path.to_str().unwrap()],
```
[the remark this answers is not in the document]
Comment (joyliu-q) [reply worker/src/rebuild.rs:142]: Sounds good!

**Comment by adityasaky** [comment]
Fixed the conflict with Cargo.lock.

File: worker/src/rebuild.rs [hunk 2]
```diff
@@ -128,12 +131,33 @@ pub async fn rebuild(ctx: &Context<'_>, url: &str) -> Result<Rebuild> {
...
     // process result
     let output_path = out_dir.join(&filename);
+
+    info!("Generating signed link");
+
+
+    let signed_link = in_toto_run(
+        &format!("rebuild {}", filename.to_str().unwrap()),
+        None,
+        &[input_path.to_str().unwrap_or_else(|| "")],
```
Comment (kpcyrd) [inline worker/src/rebuild.rs:141]:
```suggestion
        &[input_path.to_str().ok_or_else(|| anyhow!("Input path contains invalid characters"))?],
```
This converts the `Option` into a `Result` that you can then use the `?` on :) If there are invalid characters the function would then just return that error.

**Comment by kpcyrd** [comment]
I noticed the tests don't run in eg. https://github.com/in-toto/rebuilderd/pull/2, this is because they're only executed if the pull request is opened on a branch named `main`. If you open a pull request to the main branch in https://github.com/joyliu-q/rebuilderd/ you can test-run the github actions there (you don't have to merge, you can just close it after the tests pass). :)

**Comment by kpcyrd** [comment]
Thanks for all the work by everybody involved in this! :rocket: 

One of the rebuilders deployed the current main branch and is now generating attestations for Arch Linux packages:

https://wolfpit.net/rebuild/api/v0/builds/26738/attestation ([archive.org](https://web.archive.org/web/20210906232252/https://wolfpit.net/rebuild/api/v0/builds/26738/attestation))

Great work @joyliu-q! :tada:

**Pull request merged**